gapic/src/main/com/google/gapid/perfetto/views/RootPanel.java

```diff
@@ -196,6 +200,7 @@ public void render(RenderContext ctx, Repainter repainter) {
 
   protected abstract void preTopUiRender(RenderContext ctx, Repainter repainter);
   protected abstract void preMainUiRender(RenderContext ctx, Repainter repainter);
+  protected abstract void renderFlags(RenderContext ctx, Panel panel);
```

Comment:
No need as an abstract method, refer to renderVSync.

Reply:
This needs to be abstract because it is similar to the preTop and preMain render functions and not the renderVSync. The flags have to be rendered after the top and bottom have finished rendering, for it to be on top of all other events. This means calling renderFlags from within the render() of RootPanel. We don't want flags in the Replay Profiler, so this has to be abstract and only the ForSystemTrace will implement it.

Reply:
I still hold the opinion that it's closer to renderVsync than to preTop/preMain but it's not a blocker for this PR, so I'm fine with having it checked in.

- PreTop/preMain emphasize function's execution time, and what a real panel do at preTop time may indeed vary a lot so it's reasonable to declare them abstract. 
- RenderVsync/renderFlagBut emphasize the actual function of the module, if a panel decide to render flags, how they render wouldn't vary that much, so it's not that straightforward for me to mark it abstract. 

I'm not sure whether we want to toggle off flag function in replay profiler, but even if we do so, we can still override renderFlags() method to empty even if it's not declared as abstract.